v4/export/sql.go

```diff
@@ -521,7 +569,7 @@ func isUnknownSystemVariableErr(err error) bool {
 	return strings.Contains(err.Error(), "Unknown system variable")
 }
 
-func resetDBWithSessionParams(tctx *tcontext.Context, db *sql.DB, dsn string, params map[string]interface{}) (*sql.DB, error) {
+func (d *Dumper) resetDBWithSessionParams(tctx *tcontext.Context, db *sql.DB, dsn string, params map[string]interface{}) error {
```

Comment:
seems `tidb_snapshot` is included in `params` and will be set for `db`, so `oldDBHandle` is also set snapshot, where its description is "needs dbHandle without snapshot to make sure"

Reply:
`oldDBHandle` is opened without `tidb_snapshot`. First `oldDBHandle` doesn't have any `params`.